### ISSUE TITLE
Write a "configure"-style script

### ISSUE THREAD

**Issue by cgjones** [mozilla/rr#1112]
Several users have been tripped up by not having rr build deps, or not configuring their machine appropriately.  Most projects handle this by having a "configure" step that checks the build environment.

We already have a `configure` script that's used for another purpose, so it'd be relatively easy to extend that to check deps/config too.

Note: unfortunately, the rr exe will still need to duplicate these env checks because users can install prebuilt packages.

**Comment by cgjones** [comment]
Good first bug for someone who knows CMake reasonably well.

**Comment by khuey** [comment]
Long term we need to figure out if cmake can sanely support cross compiling.  I wasn't able to find much about doing that online last year, unfortunately.

**Comment by cgjones** [comment]
Seems like it can be done: http://www.cmake.org/Wiki/CMake_Cross_Compiling

**Comment by luser** [comment]
I poked at this a bit:
https://github.com/luser/rr/commit/06f3d257c850e7bd8d5483290061b9f278da274d

The only required library I saw was zlib, so I added a check for that. It got a little fiddly trying to do the right thing in the 64->32 cross compile case. I made it work for Ubuntu but I don't have Fedora handy and I know things are different there. I'll get a Fedora VM or something and try that out.

Is there anything else we ought to be checking for? Looks like you use Python to generate some files, so checking for that would be good, and maybe checking a specific version too.
